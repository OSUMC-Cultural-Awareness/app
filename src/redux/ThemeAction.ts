import { Store } from ".";

export const updateTheme = (
  theme: Store["theme"]
): { type: string; payload: Store["theme"] } => ({
  type: "UPDATE_THEME",
  payload: theme,
});
