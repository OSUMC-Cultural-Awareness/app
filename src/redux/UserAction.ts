import { Store } from ".";

export const updateUser = (
  user: Store["user"]
): { type: string; payload: Store["user"] } => ({
  type: "UPDATE_USER",
  payload: user,
});

export const resetUser = (): {
  type: string;
  payload: Record<string, unknown>;
} => ({
  type: "RESET_USER",
  payload: {},
});
