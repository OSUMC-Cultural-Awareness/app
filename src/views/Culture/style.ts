import { StyleSheet, Platform } from "react-native";

export default StyleSheet.create({
  spinner: { top: "50%", position: "relative" },

  specialAddInsight: {
    padding: 10,
    marginVertical: 5,
    marginHorizontal: 5,
  },

  view: {
    flex: 1,
  },

  card: {
    padding: 10,
    marginVertical: 5,
    marginHorizontal: 5,
  },

  // HACK: Any here so that FABs are Fixed on web, but use the default styling on Mobile.
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  fab: { position: (Platform.OS === "web" ? "fixed" : undefined) as any },
});
