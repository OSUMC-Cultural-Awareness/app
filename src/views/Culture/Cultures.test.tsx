/**
 * @jest-environment jsdom
 */
import React from "react";
import "react-native";
import { NavigationContainer } from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";

import Culture from ".";
import { USER_INITIAL_STATE, THEME_INITIAL_STATE } from "../../redux";
import { Routes } from "../../routes";

const CultureTest = () => {
  const Stack = createStackNavigator<Routes>();

  const mockStore = configureStore([]);

  return (
    <Provider
      store={mockStore({
        user: USER_INITIAL_STATE,
        theme: THEME_INITIAL_STATE,
      })}
    >
      <NavigationContainer>
        <Stack.Navigator initialRouteName="Culture">
          <Stack.Screen name="Culture" component={Culture} />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  );
};

describe("Culture", () => {
  beforeEach(() => {
    jest.useFakeTimers();
    jest.clearAllMocks();
  });

  test("Renders", () => {
    const rendered = renderer.create(<CultureTest />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
