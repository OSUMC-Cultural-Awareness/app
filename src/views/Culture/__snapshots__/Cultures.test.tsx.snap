// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Culture Renders 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "marginEnd": 25,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontFamily": "System",
                          "fontWeight": "500",
                          "textAlign": "left",
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "fontSize": 20,
                            "letterSpacing": 0.15,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          Object {
                            "fontWeight": "600",
                          },
                        ],
                      ],
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "flex-end",
                  "bottom": 0,
                  "justifyContent": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "justifyContent": "center",
                    }
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "disabled": undefined,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    hitSlop={
                      Object {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "alignItems": "center",
                            "justifyContent": "center",
                            "margin": 6,
                            "overflow": "hidden",
                          },
                          Object {
                            "borderRadius": 18,
                            "height": 36,
                            "width": 36,
                          },
                          undefined,
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View>
                      <Text />
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "justifyContent": "center",
                    }
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "disabled": undefined,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    hitSlop={
                      Object {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "alignItems": "center",
                            "justifyContent": "center",
                            "margin": 6,
                            "overflow": "hidden",
                          },
                          Object {
                            "borderRadius": 18,
                            "height": 36,
                            "width": 36,
                          },
                          undefined,
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View>
                      <Text />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessibilityRole="progressbar"
                        accessible={true}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "justifyContent": "center",
                            },
                            Object {
                              "position": "relative",
                              "top": "50%",
                            },
                          ]
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "height": 48,
                              "opacity": 1,
                              "width": 48,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "bottom": 0,
                                "justifyContent": "center",
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "height": 48,
                                  "transform": Array [
                                    Object {
                                      "rotate": "45deg",
                                    },
                                  ],
                                  "width": 48,
                                }
                              }
                            >
                              <View
                                collapsable={false}
                                style={
                                  Object {
                                    "height": 24,
                                    "overflow": "hidden",
                                    "width": 48,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "height": 48,
                                      "transform": Array [
                                        Object {
                                          "translateY": 0,
                                        },
                                        Object {
                                          "rotate": "-165deg",
                                        },
                                      ],
                                      "width": 48,
                                    }
                                  }
                                >
                                  <View
                                    collapsable={false}
                                    style={
                                      Object {
                                        "height": 24,
                                        "overflow": "hidden",
                                        "width": 48,
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "borderColor": "#6200ee",
                                          "borderRadius": 24,
                                          "borderWidth": 4.8,
                                          "height": 48,
                                          "width": 48,
                                        }
                                      }
                                    />
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "bottom": 0,
                                "justifyContent": "center",
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "height": 48,
                                  "transform": Array [
                                    Object {
                                      "rotate": "45deg",
                                    },
                                  ],
                                  "width": 48,
                                }
                              }
                            >
                              <View
                                collapsable={false}
                                style={
                                  Object {
                                    "height": 24,
                                    "overflow": "hidden",
                                    "top": 24,
                                    "width": 48,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "height": 48,
                                      "transform": Array [
                                        Object {
                                          "translateY": -24,
                                        },
                                        Object {
                                          "rotate": "345deg",
                                        },
                                      ],
                                      "width": 48,
                                    }
                                  }
                                >
                                  <View
                                    collapsable={false}
                                    style={
                                      Object {
                                        "height": 24,
                                        "overflow": "hidden",
                                        "width": 48,
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "borderColor": "#6200ee",
                                          "borderRadius": 24,
                                          "borderWidth": 4.8,
                                          "height": 48,
                                          "width": 48,
                                        }
                                      }
                                    />
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
